packages/backend/src/apps/paysg/actions/create-payment/index.ts

```diff
@@ -0,0 +1,169 @@
+  const metadata = ((parameters.metadata as IJSONObject[] | null) ?? []).filter(
+    (metadatum) => !!metadatum.key,
+  )
+  if (metadata?.length) {
+    for (const metadatum of metadata) {
+      const { key, value } = metadatum as { key: string; value: string }
+      payload['metadata'][key] = value
+    }
+  }
+
+  return payload
+}
+
+const action: IRawAction = {
+  name: 'Create Payment',
+  key: 'createPayment',
+  description: 'Create a new PaySG payment',
+  arguments: [
+    {
+      label: 'Reference ID',
+      key: 'referenceId',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Payer Name',
+      key: 'payerName',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Payer Address',
+      key: 'payerAddress',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Payer Identifier',
+      description: 'e.g. NRIC',
+      key: 'payerIdentifier',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Payer Email',
+      key: 'payerEmail',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Description',
+      key: 'description',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Payment amount (in cents)',
+      key: 'paymentAmountCents',
+      type: 'string' as const,
+      required: true,
+      variables: true,
+    },
+    {
+      label: 'Due Date',
+      description: 'Must be formatted as DD-MM-YYYY (e.g. 31-DEC-2023)',
+      key: 'dueDate',
+      type: 'string' as const,
+      required: false,
+      variables: true,
+    },
+    {
+      label: 'Return URL',
+      key: 'returnUrl',
+      type: 'string' as const,
+      required: false,
+      variables: true,
+    },
+    {
+      label: 'Metadata',
+      key: 'metadata',
+      type: 'multirow' as const,
+      required: false,
+      subFields: [
+        {
+          placeholder: 'Key',
+          key: 'key',
+          type: 'string' as const,
+          required: true,
+          variables: true,
+        },
+        {
+          placeholder: 'Value',
+          key: 'value',
+          type: 'string' as const,
+          required: true,
+          variables: true,
+        },
+      ],
+    },
+  ],
+
+  async run($) {
+    const apiKey = $.auth.data.apiKey as string
+    const baseUrl = getApiBaseUrl(apiKey)
+    const paymentServiceId = $.auth.data.paymentServiceId as string
+    const payload = constructPayload($.step.parameters)
+
+    const response = await $.http.post(
+      `/v1/payment-services/${paymentServiceId}/payments`,
+      payload,
+      {
+        baseURL: baseUrl,
+        headers: {
+          'x-api-key': apiKey,
+        },
+      },
+    )
+
```

Comment:
TODO in future: handle errors properly based on https://guide.pay.gov.sg/api/errors